src/passes/OptimizeInstructions.cpp

```diff
@@ -1204,6 +1207,29 @@ struct OptimizeInstructions
       }
     }
 
+    auto features = getModule()->features;
+
+    // Repeated identical ref.cast operations are unnecessary, if using the
+    // exact same rtt - the result will be the same. Find the immediate child
+    // cast, if there is one, and see if it is identical.
+    // TODO: Look even further through incompatible casts?
+    auto* ref = curr->ref;
+    while (!ref->is<RefCast>()) {
+      auto* last = ref;
+      ref = Properties::getImmediateFallthrough(ref, passOptions, features);
```

Comment:
Why `ImmediateFallthrough` and not just `Fallthrough`?

Reply:
Fallthrough will look through a cast! :smile: 

The meaning of fallthrough is that the value passes through, without being changed (unless we trap or branch). But here we want to ignore all other falling-through but via casting...

Reply:
Oh, that makes sense!